cloudsync/tests/test_cs.py

```diff
@@ -3034,10 +3036,50 @@ def test_smart_delete_path_local(scs):
 
     scs.run_until_clean(timeout)
 
+    # test 2: confirm that deleting a file locally removes the file remotely
     assert not remote.exists_path(remote_dir)
     assert not remote.exists_path(remote_file)
 
 
+def test_smartsync_rename(scs):
+    timeout = 1
+    local_parent = "/local"
+    remote_parent = "/remote"
+    local_test_folder = "/local/testfolder"
+    remote_test_folder = "/remote/testfolder"
+    remote_test_folder_contents = "/remote/testfolder/testfile"
+    remote_path1 = "/remote/testfolder/stuff1"
+    local_path1 = "/local/testfolder/stuff1"
+    remote_path2 = "/remote/testfolder/stuff2"
+    local_path2 = "/local/testfolder/stuff2"
+    contents = b"hello"
+
+    local, remote = scs.providers
+    rfolder_oid = remote.mkdir(remote_parent)
+
+    remote.mkdir(remote_test_folder)
+    scs.run_until_clean(timeout)
+
+    # confirm successful regular rename
+    linfo_start = local.create(local_path1, BytesIO(contents))
+    scs.run_until_clean(timeout)
+    assert local.exists_path(local_path1)
+    assert remote.exists_path(remote_path1)
+    new_oid = scs.smart_rename(LOCAL, linfo_start.oid, local_path2)
+    scs.run_until_clean(timeout)
+    assert local.exists_path(local_path2)
```

Comment:
could also be good to assert that the old paths don't exist

Reply:
done